tests/framework/session/test_session_hook_manager.py

```diff
@@ -94,22 +89,23 @@ def test_disabling_auto_discovered_hooks(
         naughty_plugin,
         good_plugin,
     ):
-        hook_manager = get_hook_manager()
-        assert not hook_manager.get_plugins()
 
         distinfo = [("plugin_obj_1", naughty_plugin), ("plugin_obj_2", good_plugin)]
-        list_distinfo_mock = mocker.patch.object(
-            hook_manager, "list_plugin_distinfo", return_value=distinfo
+        mocked_distinfo = mocker.patch(
+            "pluggy._manager.PluginManager.list_plugin_distinfo", return_value=distinfo
         )
-        mocker.patch.object(
-            hook_manager, "load_setuptools_entrypoints", return_value=len(distinfo)
+
+        mocker.patch(
+            "pluggy._manager.PluginManager.load_setuptools_entrypoints",
+            return_value=len(distinfo),
         )
-        unregister_mock = mocker.patch.object(hook_manager, "unregister")
+        unregister_mock = mocker.patch("pluggy._manager.PluginManager.unregister")
 
         KedroSession.create(
```

Comment:
This is a _very minor_ comment, but in test above we use `request.getfixturevalue("mock_session")` and here we manually create a session on the spot. Is there a meaningful different in the two approaches or should they ideally be consistent?

Reply:
Good question. So in `test_hooks_registered_when_session_created` we use the fixture because we want the session with the regular mocked settings:
```
@pytest.fixture
def mock_settings(mocker, project_hooks):
    class MockSettings(_ProjectSettings):
        _HOOKS = Validator("HOOKS", default=(project_hooks,))

    return _mock_imported_settings_paths(mocker, MockSettings())
```

In `test_disabling_auto_discovered_hooks` we want to use the `mock_settings_with_disabled_hooks` and so we create a session in the test that will then use those specific settings. I can add a comment to make that clearer.